rundeckapp/src/main/groovy/com/dtolabs/rundeck/server/plugins/logging/SimpleDataFilterPlugin.groovy

```diff
@@ -172,7 +203,17 @@ See the [Java Pattern](https://docs.oracle.com/javase/8/docs/api/java/util/regex
                 }
                 if (key && value) {
                     if(invalidKeyPattern){
-                        def validKey = key.replaceAll(invalidKeyPattern,"_")
+                        def validKey = null
+                        def emptyReplacement = invalidCharactersReplacement == null
```

Comment:
This should be inside if context
`                        if( replaceFilteredResult ){`

Reply:
'validKey' is used by different 'if' statements. (see line 215 & 218)